### DISCUSSION TITLE
Visibility and ceiling height not available?

### DISCUSSION THREAD

**Discussion by Jen-Parke** [weather-gov/api#703] [General]
Using https://api.weather.gov/gridpoints/SEW/93,100 for reference but tested on other locations.

The hourly forecast data has visibility and ceiling height entries but no data is provided (returns [] for values). Is this data implemented yet? If so, how do I get it? If not, is there any information on when it might be? The /forecast and /forecast/hourly do not have an entry for ceiling and visibility data at all.

**Comment by cgodkin** [comment]
Many readers will know this already, but visibility and ceiling are available from https://aviationweather.gov/data/api/ .  Still, it would be nice not to have to use both services.

**Reply by Jen-Parke** [reply]
Yeah, getting everything consolidated onto a single service is the goal here. But there's also a gap in the aviation coverage, unless I'm missing something? The aviationweather.gov API has METARs (current weather conditions, available at most airports) and TAFs (current and forecast conditions, but only available for some airports) but not forecast data for airports that do not have TAFs issued. So if, say, I want to know if the ceiling will be high enough to fly tomorrow morning at an airport where no TAF is issued neither API covers it.

But I take it from your suggestion that I'm not missing anything with weather.gov, the blank values for ceiling and visibility aren't just user error on my part?

**Reply by cgodkin** [reply]
No, I don't think you're missing anything.  But I do see ceiling and visibility among the "Other layers" listed on this page: https://weather-gov.github.io/api/gridpoints so maybe there's some intent to one day add them?

**Reply by sullynole** [reply]
The forecastGridData url has ceiling and visibility if the office populates those grids.

**Reply by cgodkin** [reply]
Thanks!  I didn't try that endpoint previously.

**Reply by cgodkin** [reply]
I found that BOX (Boston) apparently populates this.  Right now you can try this: https://api.weather.gov/gridpoints/BOX/73,90 and see the values that @sullynole mentions.
